### PR TITLE
Support cancelling cover movement when initiated outside Home Assistant

### PR DESCRIPTION
**Pull request by iMicknl**
(not happy with the readability of the code yet, thus marking it as a draft)

### CODE REVIEW

**Comment by iMicknl** [comment]
Functionality works, as confirmed [here](https://github.com/iMicknl/ha-tahoma/issues/283#issuecomment-703302427).

File: custom_components/tahoma/cover.py [hunk 1]
```diff
@@ -292,10 +292,27 @@ async def async_cancel_or_stop_cover(self, cancel_commands, stop_commands) -> No
...
+                and command.get("name") in cancel_commands
+            ),
+            None,
+        )
+
+        if exec_id:
+            return await self.async_cancel_command(exec_id)
+
+        # Fallback to available stop commands when no execution can be found
+        await self.async_execute_command(self.select_command(*stop_commands))
```
Comment (tetienne) [inline custom_components/tahoma/cover.py:315]: Finally, why not keep only this one? Do you think some device does not have it?

Comment (iMicknl) [reply custom_components/tahoma/cover.py:315]: You mean, removing the other logic in favour of this one? Unfortunately the stop command don't work for every device. Even when a device has a stop command (like my pergola), it just doesn't work.

Cancelling a running execution is the method to stop a cover movement (also on tahomalink.com) and it is a reliable way that works with every device. This is some Somfy logic that I couldn't work around unfortunately and it was causing quite some user issues.

**Review by vlebourl — APPROVED** [review]

File: custom_components/tahoma/cover.py (hunk 1 above)
Comment (iMicknl) [reply custom_components/tahoma/cover.py:318]: @tetienne good point by the way. I updated the documentation to make this design choice more clear.

**Review by tetienne — APPROVED** [review]

**Pull request merged**